tests/macros/t7875.nim

```diff
@@ -0,0 +1,22 @@
+discard """
+  nimout: "var mysym`gensym0: MyType[float32]"
+  joinable: false
```

Comment:
nit: why joinable false? this could be moved to a block in tmacros.nim (and with
```nim
doAssert result.repr == "var mysym`gensym0: MyType[float32]"
```